### PR TITLE
Feature: write band energy of T, VL, VNL, Hartree and XC terms separately

### PR DESCRIPTION
**Pull request by maki49**
@AroundPeking 

Usage:  set `out_eband_terms 1` in the INPUT file.
The band energy of T, VL, VNL, Hartree and XC (including EXX) terms will be written into the following files in "OUT.suffix/" with the same format as "vxc_out":
- kinetic_out
- vpp_local_out
- vpp_nonlocal_out
- vhartree_out

### CODE REVIEW

File: docs/advanced/input_files/input-main.md [hunk 1]
```diff
@@ -147,6 +147,7 @@
     - [out\_mat\_t](#out_mat_t)
     - [out\_mat\_dh](#out_mat_dh)
     - [out\_mat\_xc](#out_mat_xc)
+    - [out\_eband\_separate\_term](#out_eband_separate_term)
```
Comment (mohanchen) [inline docs/advanced/input_files/input-main.md]: the name is too long

File: docs/advanced/input_files/input-main.md [hunk 2]
```diff
@@ -1678,11 +1679,18 @@ These variables are used to control the output of properties.
 ### out_mat_xc
 
 - **Type**: Boolean
-- **Availability**: Numerical atomic orbital basis
+- **Availability**: Numerical atomic orbital (NAO) and NAO-in-PW basis
 - **Description**: Whether to print the upper triangular part of the exchange-correlation matrices in **Kohn-Sham orbital representation** (unit: Ry): $\braket{\psi_i|V_\text{xc}^\text{(semi-)local}+V_\text{exx}+V_\text{DFTU}|\psi_j}$ for each k point into files in the directory `OUT.${suffix}`, which is useful for the subsequent GW calculation. (Note that currently DeePKS term is not included. ) The files are named `k-$k-Vxc`, the meaning of `$k`corresponding to k point and spin  is same as [hs_matrix.md](../elec_properties/hs_matrix.md#out_mat_hs).
 The band (KS orbital) energy for each (k-point, spin, band) will be printed in the file `OUT.${suffix}/vxc_out`. If EXX is calculated, the local and EXX part of band energy will also be printed in `OUT.${suffix}/vxc_local_out`and `OUT.${suffix}/vxc_exx_out`, respectively. All the `vxc*_out` files contains 3 integers (nk, nspin, nband) followed by nk\*nspin\*nband lines of energy Hartree and eV.
```
Comment (mohanchen) [inline docs/advanced/input_files/input-main.md]: I would suggest adding ".dat" after these output file names

File: source/module_esolver/esolver_ks_lcao.cpp [hunk 3]
```diff
@@ -11,6 +11,7 @@
 #include "module_io/to_qo.h"
 #include "module_io/write_HS.h"
 #include "module_io/write_Vxc.hpp"
+#include "module_io/write_eband_separate_term.hpp"
```
Comment (mohanchen) [inline source/module_esolver/esolver_ks_lcao.cpp]: the file name is too long

File: source/module_io/write_eband_separate_term.hpp [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+#define _WRITE_EBAND_SEPARATE_TERMS_HPP_
+#include "module_io/write_Vxc.hpp"
+#include "module_hamilt_lcao/hamilt_lcaodft/operator_lcao/ekinetic_new.h"
+#include "module_hamilt_lcao/hamilt_lcaodft/operator_lcao/nonlocal_new.h"
+#include "module_basis/module_nao/two_center_bundle.h"
+namespace ModuleIO
+{
+    template <typename TK, typename TR>
+    void write_eband_separate_term(int nspin,
+        int nbasis,
```
Comment (mohanchen) [inline source/module_io/write_eband_separate_term.hpp]: add const be fore int if possible

File: docs/advanced/input_files/input-main.md (hunk 2 above)
Comment (maki49) [reply docs/advanced/input_files/input-main.md]: The names "xxx_out" are consistent with "vxc_out" read by LibRPA. @minyez.

Comment (minyez) [reply docs/advanced/input_files/input-main.md]: It is completely fine for LibRPA to adapt the names of inputs or use symlink. If adding ".dat" makes it more consistent with other outputs in abacus, I would also suggest adding it.

File: docs/advanced/input_files/input-main.md (hunk 1 above)
Comment (maki49) [reply docs/advanced/input_files/input-main.md]: "separate_term" is replaced by "terms" in the newest commit.

File: docs/advanced/input_files/input-main.md (hunk 2 above)
Comment (maki49) [reply docs/advanced/input_files/input-main.md]: ".dat" is added after all the "${term}_out" files in the newest commit.

File: docs/advanced/input_files/input-main.md [hunk 5]
```diff
@@ -147,6 +147,7 @@
     - [out\_mat\_t](#out_mat_t)
     - [out\_mat\_dh](#out_mat_dh)
     - [out\_mat\_xc](#out_mat_xc)
+    - [out\_eband\_separate\_term](#out_eband_terms)
```
Comment (mohanchen) [inline docs/advanced/input_files/input-main.md]: change this to new name?

File: source/module_io/write_Vxc.hpp
Comment (mohanchen) [inline source/module_io/write_Vxc.hpp:1]: can you change V to v?

**Review by mohanchen — APPROVED** [review]

**Pull request merged**